airbyte-ci/connectors/metadata_service/orchestrator/orchestrator/assets/registry.py

```diff
@@ -11,13 +11,49 @@
 from google.cloud import storage
 from metadata_service.models.generated.ConnectorRegistryV0 import ConnectorRegistryV0
 from metadata_service.models.transform import to_json_sanitized_dict
-from orchestrator.assets.registry_entry import read_registry_entry_blob
+from orchestrator.assets.registry_entry import ConnectorTypePrimaryKey, ConnectorTypes, read_registry_entry_blob
 from orchestrator.logging import sentry
 from orchestrator.logging.publish_connector_lifecycle import PublishConnectorLifecycle, PublishConnectorLifecycleStage, StageStatus
+from orchestrator.utils.object_helpers import default_none_to_dict
+from pydash.objects import set_with
 
 GROUP_NAME = "registry"
 
 
+def _get_registry_entry_id(registry_entry: dict, connector_type: ConnectorTypes) -> str:
+    """Get the registry entry ID.
```

Comment:
💅 It would have helped me figured this helper behavior if the docstring mentionned the pk_field are sourceDefinitionId or destinationDefinitionId